### ISSUE TITLE
Support for events on Polarchart (for Pie.Chart)

### ISSUE THREAD

**Issue by nestorLanex2** [FormidableLabs/victory-native-xl#252]
Good Day! 

I would like to ask if _**events**_, or a similar implementation to it will be supported on the Pie.Chart in the future? 
currently I'm on 35.5.5 and thus using VictoryPie to display the pie chart component. And on the latest versions I believe _**PolarChart**_ with child **<Pie.Chart>** is the newer implementation. but events or gestures seems not to be supported yet. 

I'm hesitant to upgrade to the latest version because of this.

**Comment by zibs** [comment]
You're correct in that it's not supported at the moment, but it's definitely on the roadmap -- what kind of use case do you have in mind?

**Comment by nestorLanex2** [comment]
@zibs in the previous versions that support the Victory Pie component, it's possible to click on the slices of the pie and trigger an event. [Something like this](https://commerce.nearform.com/open-source/victory/docs/victory-pie#events)

Currently our app utilizes this use case, and since the newer libraries do not support this case yet, We'll have to stick with the latest version still supporting Victory Pie.

**Comment by Than-DE** [comment]
Same here, will need this feature to do something similar like the "Buddy" app. When clicking a pie (donut) slice, it should show the information of the slice in the middle.

![CleanShot 2024-05-10 at 09 54 39@2x](https://github.com/FormidableLabs/victory-native-xl/assets/74593656/5a6b8dbc-1ba8-472b-88a6-3c89fe66676e)

@nestorLanex2 what is your experience like using two different versions of Victory Charts?

**Comment by nestorLanex2** [comment]
@Than-DE 
On my use case, There was a need to upgrade the packages on our end since we're upgrading our RN version to LTS. When I tried upgrading to the latest version of the Pie chart, it still rendered the ui correctly but it just lacked the feature to handle events when clicking on the slices, thus I had to revert to the LTS version that still supported the <VictoryPie> component. I haven't used it extensively to give further insight on the libraries as a whole but what I could say is that as soon as the behavior will be supported in the latest versions, I'll readily upgrade to the latest right away.

**Comment by maksym-kyryliuk** [comment]
@zibs Any updates on this feature? At least a click on a pie slice would be useful, like for selecting a slice and changing its size.

**Comment by maksym-kyryliuk** [comment]
@zibs Any updates on this feature? At least a click on a pie slice would be useful, like for selecting a slice and changing its size.

**Comment by iamhassanraza** [comment]
Any update on this feature? I kinda went all in with victory native and then at this point it became a limation. Any work around to achieve this ?

**Comment by kimchouard** [comment]
+1 on this one. Any news on this? It was on the roadmap more than a year ago, so hoping it'll be available soon? 👀

The ability to click on charts to filter is a pretty basic and important feature. This library is great and has a lot of potential, good chart libs for RN are rare... So not having this feature is a real bummer.

@zibs what's missing to have this? Any help we can provide on the implementation? Happy to work on a quick PR if provided with some high level guidance 🤓

**Comment by samuelbadru** [comment]
Also want to add momentum to this issue, as it's something that there's clearly demand for. In my case I want to display a tooltip when a user presses on a pie slice. Having parody with Cartesian Charts with a [chartPressState prop](https://nearform.com/open-source/victory-native/docs/cartesian/chart-gestures#the-chartpressstate-prop-of-cartesianchart) would be great.
